pkg/dispatchcli/cmd/create_function.go

```diff
@@ -28,6 +28,7 @@ var (
 	schemaInFile          = ""
 	schemaOutFile         = ""
 	fnSecrets             = []string{}
+	fnServices            = []string{}
```

Comment:
You are defining this to be an empty slice, meaning if a user does not provide anything, it will still be sent do API as `[]`, not as nil. Later in API code you are checking if `services == nil` (which should be done in case someone is using API directly). That's ok in general, but it caused problems before when empty slice defined in subscription CLI would overwrite non-empty secrets defined for function. Might be worth checking if the empty slice is really what you meant here.

Reply:
ok.  I was more or less just copying the secrets impl.  I'll check.